src/Compilers/CSharp/Portable/Parser/Lexer_RawStringLiteral.cs

```diff
@@ -0,0 +1,347 @@
+                info.StringValue = "";
+
+            Debug.Assert(info.StringValue != null);
+            info.Text = TextWindow.GetText(intern: true);
+        }
+
+        private void ScanSingleLineRawStringLiteral(ref TokenInfo info, int startingQuoteCount)
+        {
+            info.Kind = SyntaxKind.SingleLineRawStringLiteralToken;
+
+            while (true)
+            {
+                var currentChar = TextWindow.PeekChar();
+
+                // See if we reached the end of the line or file before hitting the end.
+                if (SyntaxFacts.IsNewLine(currentChar))
+                {
+                    this.AddError(TextWindow.Position, width: GetNewLineWidth(currentChar), ErrorCode.ERR_Unterminated_raw_string_literal);
+                    return;
+                }
+                else if (IsAtEndOfText(currentChar))
+                {
+                    this.AddError(TextWindow.Position, width: 0, ErrorCode.ERR_Unterminated_raw_string_literal);
+                    return;
+                }
+
+                if (currentChar != '"')
+                {
+                    // anything not a quote sequence just moves it forward.
+                    TextWindow.AdvanceChar();
+                    continue;
+                }
+
+                var beforeEndDelimeter = TextWindow.Position;
+                var currentQuoteCount = ConsumeQuoteSequence();
+
+                // A raw string literal starting with some number of quotes can contain a quote sequence with less quotes.
+                if (currentQuoteCount < startingQuoteCount)
+                    continue;
+
+                // A raw string could never be followed by another string.  So once we've consumed all the closing quotes
+                // if we have any more closing quotes then that's an error we can give a message for.
+                if (currentQuoteCount > startingQuoteCount)
+                {
+                    var excessQuoteCount = currentQuoteCount - startingQuoteCount;
+                    this.AddError(
+                        position: TextWindow.Position - excessQuoteCount,
+                        width: excessQuoteCount,
+                        ErrorCode.ERR_Too_many_closing_quotes_for_raw_string_literal);
+                }
+
+                // We have enough quotes to finish this string at this point.
+                var afterStartDelimeter = TextWindow.LexemeStartPosition + startingQuoteCount;
+                var valueLength = beforeEndDelimeter - afterStartDelimeter;
+
+                info.StringValue = TextWindow.GetText(
+                    position: afterStartDelimeter,
+                    length: valueLength,
+                    intern: true);
+                return;
+            }
+        }
+
+        private int GetNewLineWidth(char currentChar)
+        {
+            Debug.Assert(SyntaxFacts.IsNewLine(currentChar));
+            return currentChar == '\r' && TextWindow.PeekChar(1) == '\n' ? 2 : 1;
```

Comment:
`IsNewLine` recognizes other sorts of newline characters as well. Just want to confirm that `\r\n` is the combination that works as a pair. Is that correct?

```
            // new-line-character:
            //   Carriage return character (U+000D)
            //   Line feed character (U+000A)
            //   Next line character (U+0085)
            //   Line separator character (U+2028)
            //   Paragraph separator character (U+2029)
```

Reply:
> Just want to confirm that \r\n is the combination that works as a pair. Is that correct?

correct.  it's the only case that we consume 2 chars into one effective newline.

Reply:
#resolved